src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/VariantsToTable.java

```diff
@@ -0,0 +1,430 @@
+        oneLineSummary = "Extract specific fields from a VCF file to a tab-delimited table",
+        programGroup = VariantProgramGroup.class
+)
+public final class VariantsToTable extends VariantWalker {
+
+    @Argument(fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME,
+            shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME,
+            doc="File to which results should be written")
+    private String out = null;
+
+    /**
+     * -F NAME can be any standard VCF column (CHROM, ID, QUAL) or any binding in the INFO field (e.g., AC=10).
+     * Note that to capture GENOTYPE (FORMAT) field values, see the GF argument.  This argument accepts any number
+     * of inputs.  So -F CHROM -F POS is allowed.
+     */
+    @Argument(fullName="fields", shortName="F", doc="The name of each field to capture for output in the table", optional=true)
+    private List<String> fieldsToTake = new ArrayList<>();
+
+    /**
+     * -GF NAME can be any binding in the FORMAT field (e.g., GQ, PL).
+     * Note this argument accepts any number of inputs.  So -GF GQ -GF PL is allowed.
+     */
+    @Argument(fullName="genotypeFields", shortName="GF", doc="The name of each genotype field to capture for output in the table", optional=true)
+    private List<String> genotypeFieldsToTake = new ArrayList<>();
+
+    /**
+     * By default this tool only emits values for fields where the FILTER field is either PASS or . (unfiltered).
+     * Throwing this flag will cause VariantsToTable to emit values regardless of the FILTER field value.
+     */
+    @Advanced
+    @Argument(fullName="showFiltered", shortName="raw", doc="If provided, field values from filtered records will be included in the output", optional=true)
+    private boolean showFiltered = false;
+
+    /**
+     * By default, records with multiple ALT alleles will comprise just one line of output; note that in general this can make your resulting file
+     * unreadable/malformed for certain tools like R, as the representation of multi-allelic INFO field values are often comma-separated lists
+     * of values.  Using the flag will cause multi-allelic records to be split into multiple lines of output (one for each allele in the ALT field);
+     * INFO field values that are not lists are copied for each of the output records while only the appropriate entry is used for lists.
+     */
+    @Argument(fullName="splitMultiAllelic", shortName="SMA", doc="If provided, we will split multi-allelic records into multiple lines of output", optional=true)
+    private boolean splitMultiAllelic = false;
+
+    /**
+     * By default, this tool emits one line per usable VCF record (or per allele if the -SMA flag is provided).  Using the -moltenize flag
+     * will cause records to be split into multiple lines of output: one for each field provided with -F or one for each combination of sample
+     * and field provided with -GF.  Note that the "Sample" column for -F fields will always be "site".
```

Comment:
I think this means that in moltenized output, all rows will have 'site' as the value of the 'Sample' column (check out the moltenized output in this pull request)